langgraph_bot/tools/tools.py

```
@@ -27,3 +35,131 @@ def title_tool(posttitle:str,postdata:str):
+        papers.append({
+            "title": d.metadata.get("title"),
+            "authors": d.metadata.get("authors"),
+            "published": d.metadata.get("published"),
+            "abstract": d.page_content,
+            "pdf_url": d.metadata.get("source"),
+        })
+
+    return papers
+
+@tool
+def scraper_tool(url)-> List:   # for static html rich content scrapping or page loading. 
```

Ruff (0.14.10)

Unused function argument: `url`

(ARG001)